Signal/src/view controllers/MessageComposeTableViewController.m

```diff
@@ -24,24 +20,44 @@ @interface MessageComposeTableViewController () <UISearchBarDelegate,
                                                  UISearchResultsUpdating,
                                                  MFMessageComposeViewControllerDelegate>
 
-@property (nonatomic, strong) IBOutlet UITableViewCell *inviteCell;
-@property (nonatomic, strong) IBOutlet OWSNoSignalContactsView *noSignalContactsView;
+@property (nonatomic) IBOutlet UITableViewCell *inviteCell;
+@property (nonatomic) UITableViewCell *conversationForNonContactCell;
+@property (nonatomic) UITableViewCell *inviteViaSMSCell;
+@property (nonatomic) IBOutlet OWSNoSignalContactsView *noSignalContactsView;
 
-@property (nonatomic) UIButton *sendTextButton;
-@property (nonatomic, strong) UISearchController *searchController;
-@property (nonatomic, strong) UIActivityIndicatorView *activityIndicator;
-@property (nonatomic, strong) UIBarButtonItem *addGroup;
-@property (nonatomic, strong) UIView *loadingBackgroundView;
+@property (nonatomic) UISearchController *searchController;
+@property (nonatomic) UIActivityIndicatorView *activityIndicator;
+@property (nonatomic) UIBarButtonItem *addGroup;
+@property (nonatomic) UIView *loadingBackgroundView;
 
 @property (nonatomic) NSString *currentSearchTerm;
 @property (copy) NSArray<Contact *> *contacts;
 @property (copy) NSArray<Contact *> *searchResults;
 @property (nonatomic, readonly) OWSContactsManager *contactsManager;
 
+// This property should be set IFF the current search text can
+// be parsed as a phone number. If set, it contains a E164 value.
+@property (nonatomic) NSString *searchPhoneNumber;
```

Comment:
Nit: So as to communicate to the caller that it's a potentially valid signal id (e164 formatted phone number), what about "searchSignalId"? (I doubled back here to check that we weren't sending a formatted phone number)